src/Components/Artwork/SaveButton/SaveArtworkToListsButton.tsx

```diff
@@ -18,79 +15,19 @@ const SaveArtworkToListsButton: FC<SaveArtworkToListsButtonProps> = ({
   artwork,
   contextModule,
 }) => {
-  const tracking = useTracking()
-  const { t } = useTranslation()
-  const { sendToast } = useToasts()
-  const { state, savedListId, dispatch } = useManageArtworkForSavesContext()
-
-  const customListsCount = artwork.customCollections?.totalCount ?? 0
-  const hasCustomLists = customListsCount > 0
-  const isSaved = !!artwork.is_saved || hasCustomLists
-
-  const openManageArtworkForSavesModal = () => {
-    dispatch({
-      type: "SET_ARTWORK",
-      payload: {
-        id: artwork.id,
-        internalID: artwork.internalID,
-        title: `${artwork.title}, ${artwork.date}`,
-        imageURL: artwork.preview?.url ?? null,
-      },
-    })
-  }
-
-  const showToastByAction = (action: string) => {
-    if (action === "Saved Artwork") {
-      sendToast({
-        variant: "success",
-        message: t(
-          "collectorSaves.saveArtworkToListsButton.artworkSavedToast.message"
-        ),
-        action: {
-          label: t(
-            "collectorSaves.saveArtworkToListsButton.artworkSavedToast.button"
-          ),
-          onClick: openManageArtworkForSavesModal,
-        },
-      })
-
-      return
-    }
-
-    sendToast({
-      variant: "message",
-      message: t(
-        "collectorSaves.saveArtworkToListsButton.artworkRemovedToast.message"
-      ),
-    })
-  }
-
-  const { handleSave } = useSaveArtwork({
-    isSaved,
+  const { state, savedListId } = useManageArtworkForSavesContext()
+  const { isSaved, onSave } = useSaveArtworkToLists({
```

Comment:
what do you think, maybe instead of using a custom hook, use [this approach](https://github.com/artsy/force/commit/e909c3054bccd1bd887cdf5ae4bfaea02b6fce55)? Advantages of this approach:
* all the logic is in one place
* all required query fields are requested in one place
* no duplication of logic

Disadvantages of the approach:
* it is more difficult to understand the code
*

Reply:
Makes sense to me to have a hook, in order to share this behavior between the artwork detail actions as well as the artwork grid brick.